hail/src/main/scala/is/hail/expr/LowerArrayToStream.scala

```diff
@@ -0,0 +1,87 @@
+package is.hail.expr.ir
+
+import is.hail.expr.types.virtual.{TArray, TContainer, TStream, TStreamable}
+
+object LowerArrayToStream {
+  private def boundary(node: IR): IR = {
+    var streamified = streamify(node)
+
+    if (streamified.typ.isInstanceOf[TStream] && node.typ.isInstanceOf[TArray]) {
+      println("Wrapping")
+      streamified = ToArray(streamified)
+    }
+
+    if (streamified.typ.isInstanceOf[TArray] && node.typ.isInstanceOf[TStream])
+      streamified = ToStream(streamified)
+
+    assert(streamified.typ == node.typ)
+    streamified
+  }
+
+  private def toStream(node: IR): IR = {
+    assert(node.typ.isInstanceOf[TStreamable])
```

Comment:
I think this should be:
```
val streamified = streamify(node)
if (streamified.typ.isInstanceOf[TStream])
  streamified
else
  ToStream(streamified)
```

will that work? Then you don't need `toStream(streamify(...))` below